src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1626 @@
+        private int I2cMAxBuffer = 1024;
+        private ParametersFlags _parametersFlags;
+
+        private SpiDevice _spiDevice = null;
+        private I2cDevice _i2cDevice = null;
+        private GpioController _controller = null;
+        private SerialPort _serialPort = null;
+        private int _pin = 18;
+        private SecurityAccessModuleMode _securityAccessModuleMode;
+        private uint _virtualCardTimeout = 20;
+
+        /// <summary>
+        /// Set or get the read timeout
+        /// Please refer to the documentation to set the right
+        /// timeout value depending on the communication
+        /// mode you are using
+        /// </summary>
+        public int ReadTimeOut { get; set; } = 500;
+
+        /// <summary>
+        /// The Log level
+        /// </summary>        
+        public LogLevel LogLevel { get { return LogInfo.LogLevel; } set { LogInfo.LogLevel = value; } }
+
+        /// <summary>
+        /// The location to log the info
+        /// </summary>
+        public LogTo LogTo { get { return LogInfo.LogTo; } set { LogInfo.LogTo = value; } }
+
+        /// <summary>
+        /// Firmware version information
+        /// </summary>
+        public FirmwareVersion FirmwareVersion { get; internal set; }
+
+        #region Spi and I2c Settings
+        /// <summary>
+        /// PN532 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 2_000_000;
+
+        /// <summary>
+        /// Only SPI Mode supported is Mode0
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The default I2C address
+        /// </summary>
+        public const byte I2cDefaultAddress = 0x24;
+
+        #endregion        
+
+        /// <summary>
+        /// Create a PN532 using Serial Port
+        /// </summary>
+        /// <param name="portName">The port name</param>
+        public Pn532(string portName, LogLevel logLevel = LogLevel.None)
+        {
+            LogLevel = logLevel;
+
+            // Data bit : 8 bits,
+            // Parity bit : none,
+            // Stop bit : 1 bit,
+            // Baud rate : 115 200 bauds,
+            LogInfo.Log("Opening serial port 115200, Parity.None, 8 bits, StopBits.One", LogLevel.Debug);
+            _serialPort = new SerialPort(portName, 115200, Parity.None, 8, StopBits.One);
+            _serialPort.ReadTimeout = SerialReadTimeOut;
+            _serialPort.WriteTimeout = SerialReadTimeOut;
+            _serialPort.Open();
+            // Setting up internals as default version
+            _parametersFlags = ParametersFlags.AutomaticATR_RES | ParametersFlags.AutomaticRATS | ParametersFlags.ISO14443_4_PICC;
+            _securityAccessModuleMode = SecurityAccessModuleMode.Normal;
+            _virtualCardTimeout = 20;
+
+            WakeUp();
+            // Set the SAM
+            bool ret = SetSecurityAccessModule();
+            //var ret = SetSAM();
```

Comment:
why is this commented out? is there some implied TODO here?

Reply:
Just forgot to remove it